synapse/storage/push_rule.py

```diff
@@ -56,6 +56,47 @@ def get_push_rules_enabled_for_user(self, user_name):
         })
 
     @defer.inlineCallbacks
+    def bulk_get_push_rules(self, user_ids):
+        batch_size = 100
+
+        def f(txn, user_ids_to_fetch):
+            sql = (
+                "SELECT " +
+                ",".join(map(lambda x: "pr."+x, PushRuleTable.fields)) +
+                " FROM " + PushRuleTable.table_name + " pr " +
+                " LEFT JOIN " + PushRuleEnableTable.table_name + " pre " +
+                " ON pr.user_name = pre.user_name and pr.rule_id = pre.rule_id " +
+                " WHERE pr.user_name " +
+                " IN (" + ",".join(["?" for _ in user_ids_to_fetch]) + ")"
```

Comment:
Use ``map(lambda`` or use a list comprehension, but probably not both in the same expression.
We've generally used list comprehensions in the rest of the code.
I'd drop the `[]` inside the join, e.g:

    join("?" for _ in user_ids_to_fetch)